File: clients/typescript/src/GitCommit.ts

export class GitCommit {
  private _oid: string;
  private _header: string;
  private _message: string;
  private _ts: number;

  constructor(input: {
    oid: string,
    header: string,
    message: string,
    ts: number,
  }) {
    this._oid = input.oid;
    this._header = input.header;
    this._message = input.message;
    this._ts = input.ts;
  }

  get oid(): string { return this._oid; }
  set oid(oid: string) { this._oid = oid; }

  get header(): string { return this._header; }
  set header(header: string) { this._header = header; }

  get message(): string { return this._message; }
  set message(message: string) { this._message = message; }

  get ts(): number { return this._ts; }
  set ts(ts: number) { this._ts = ts; }
}

File: clients/rust/src/git_commit.rs
// GitCommit represents a GitCommit model.
#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct GitCommit {
    #[serde(rename="oid")]
    pub oid: String,
    #[serde(rename="header")]
    pub header: String,
    #[serde(rename="message")]
    pub message: String,
    #[serde(rename="ts")]
    pub ts: i64,
}

impl GitCommit {
    pub fn new(oid: String, header: String, message: String, ts: i64) -> GitCommit {
        GitCommit {
            oid,
            header,
            message,
            ts,
        }
    }
}
